OutOfSchool/OutOfSchool.BusinessLogic/Util/MappingProfile.cs

```diff
@@ -865,6 +866,33 @@ public MappingProfile()
                 .ForMember(dest => dest.IsDeleted, opt => opt.Ignore());        
         
         CreateMap<Position, PositionDto>();
+
+        CreateMap<StudySubjectCreateUpdateDto, StudySubject>()
+            .ForMember(dest => dest.CreatedAt, opt => opt.Ignore())
+            .ForMember(dest => dest.CreatedBy, opt => opt.Ignore())
+            .ForMember(dest => dest.ModifiedBy, opt => opt.Ignore())
+            .ForMember(dest => dest.UpdatedAt, opt => opt.Ignore())
+            .ForMember(dest => dest.DeletedBy, opt => opt.Ignore())
+            .ForMember(dest => dest.DeleteDate, opt => opt.Ignore())
+            .ForMember(dest => dest.IsSystemProtected, opt => opt.Ignore())
+            .ForMember(dest => dest.IsDeleted, opt => opt.Ignore())
+            .ForMember(dest => dest.Document, opt => opt.Ignore())
+            .ForMember(dest => dest.File, opt => opt.Ignore())
+            .ForMember(dest => dest.ActiveFrom, opt => opt.Ignore())
+            .ForMember(dest => dest.ActiveTo, opt => opt.Ignore())
+            .ForMember(dest => dest.IsBlocked, opt => opt.Ignore())
+            .ForMember(dest => dest.PrimaryLanguage, opt => opt.Ignore())
+            .ForMember(dest => dest.Languages,
+                opt => opt.Ignore())
+            .ForMember(dest => dest.PrimaryLanguageId,
+                opt => opt.MapFrom(src => src.LanguagesSelection.FirstOrDefault(l => l.IsPrimary).Id));
+
+        CreateMap<StudySubject, StudySubjectDto>()
+            .ForMember(dest => dest.WorkshopId, opt => opt.Ignore())
+            .ForMember(dest => dest.Languages,
+                opt => opt.MapFrom(src => src.Languages.Select(l => new Language { Id = l.Id, Code = l.Code, Name = l.Name } )));
```

Comment:
you are mapping from entity to dto, are you sure `new Language` is the correct object? Also, you already have `CreateMap<Language, LanguageDto>().ReverseMap();` so this field should be mapped automatically, no need to specify anything in this case.